### PR TITLE
Feature: Add links by link definition

### PR DESCRIPTION
**Pull request by Hundraw**
Add feature that can add new link.

### CODE REVIEW

File: src/models/Component.js [hunk 1]
```diff
@@ -110,6 +110,24 @@ class Component extends FileInformation {
...
   }
+
+  setLinkAttribute(link) {
+    const attributeDefinition = this.definition.definedAttributes
+      .find(({ name }) => name === link.definition.attributeRef);
+    const attribute = this.attributes.find(({ definition }) => definition.type === 'Link'
+      && attributeDefinition.name === definition.name);
+
+    if (attribute) {
+      attribute.value.push(link.target);
```
Comment (Zorin95670) [inline src/models/Component.js:121]:
```js
if (attribute && !attribute.value.includes(link.target) {
  attribute.value.push(link.target);
} else if(!attribute) {
  (...)
}
```

**Review by Zorin95670 — CHANGES_REQUESTED** [review]

File: src/models/Component.js (hunk 1 above)
Comment (Hundraw) [reply src/models/Component.js:125]: done

File: tests/unit/models/Component.spec.js [hunk 2]
```diff
@@ -260,5 +261,90 @@ describe('Test class: Component', () => {
...
+          ],
+          definition: new ComponentDefinition({
+            definedAttributes: [
+              new ComponentAttributeDefinition({ name: 'name01' }),
+            ],
+          }),
+        });
+
+        component.setLinkAttribute(link);
+        expect(component.attributes[0].value[0]).toEqual('target01');
```
Comment (Zorin95670) [inline tests/unit/models/Component.spec.js:290]:
```js
expect(component.attributes[0].value).toEqual(['target01']);
```

File: tests/unit/models/Component.spec.js [hunk 3]
```diff
@@ -260,5 +261,90 @@ describe('Test class: Component', () => {
...
+          definition: new ComponentDefinition({
+            definedAttributes: [
+              new ComponentAttributeDefinition({ name: 'name01' }),
+            ],
+          }),
+        });
+
+        component.setLinkAttribute(link);
+        expect(component.attributes.length).toEqual(2);
+        expect(component.attributes[0].definition.name).toEqual('name02');
```
Comment (Zorin95670) [inline tests/unit/models/Component.spec.js:318]: check value of attribute too

File: tests/unit/models/Component.spec.js [hunk 4]
```diff
@@ -260,5 +261,90 @@ describe('Test class: Component', () => {
...
+          definition: new ComponentDefinition({
+            definedAttributes: [
+              new ComponentAttributeDefinition({ name: 'name03' }),
+            ],
+          }),
+        });
+
+        component.setLinkAttribute(link);
+        expect(component.attributes.length).toEqual(1);
+        expect(component.attributes[0].definition.name).toEqual('name03');
```
Comment (Zorin95670) [inline tests/unit/models/Component.spec.js:346]: check attribute value

**Review by Zorin95670 — CHANGES_REQUESTED** [review]

File: tests/unit/models/Component.spec.js (hunk 2 above)
Comment (Hundraw) [reply tests/unit/models/Component.spec.js:290]: done

File: tests/unit/models/Component.spec.js (hunk 3 above)
Comment (Hundraw) [reply tests/unit/models/Component.spec.js:318]: done

**Review by Zorin95670 — APPROVED** [review]

**Review by Zorin95670 — APPROVED** [review]

**Pull request merged**